js/client-teambuilder.js

```diff
@@ -1047,7 +1047,11 @@
 			};
 			buf += '<span class="detailcell detailcell-first"><label>Level</label>' + (set.level || 100) + '</span>';
 			if (this.curTeam.gen > 1) {
-				buf += '<span class="detailcell"><label>Gender</label>' + GenderChart[template.gender || set.gender || 'N'] + '</span>';
+				if (this.curTeam.format !== 'balancedhackmons') {
+					buf += '<span class="detailcell"><label>Gender</label>' + GenderChart[template.gender || set.gender || 'N'] + '</span>';
+				} else {
+					buf += '<span class="detailcell"><label>Gender</label>' + GenderChart[set.gender || 'N' || template.gender] + '</span>';
```

Comment:
Okay, we do. Next up, `set.gender` should be reset if necessary when changing templates, and `set.gender` should be in front no matter. So this should always be `set.gender || template.gender || 'N'`.

Note that `set.gender || 'N' || template.gender` just means `set.gender || 'N'`, since `'N'` is never falsy.